numba/tests/test_struct_ref.py

```diff
@@ -226,6 +229,14 @@ def test_MyStructType_in_dict_mixed_type_error(self):
             # the second field is now an integer.
             td['b'] = MyStruct(2.3, 1)
 
+    def test_MyStructType_hash_no_typeof_recursion(self):
+        '''Tests that __hash__ is not called prematurely in typeof 
+           causing infinite recursion (see #8241).'''
```

Comment:
```suggestion
        # Tests that __hash__ is not called prematurely in typeof
        # causing infinite recursion (see #8241).
```
docstrings end up polluting Numba's test runner output in verbose mode (also fixes `flake8` issue).